src/first-route.js

```diff
@@ -1,30 +1,54 @@
 /**
- * 
- * @param {import("fastify").FastifyInstance} fastify 
- * @param {Object} optinos 
+ *
+ * @param {import("fastify").FastifyInstance} fastify
+ * @param {Object} options
  */
 async function routes(fastify, optinos) {
-    fastify.get('/animals', async (req, res) => {
-        return fastify.prisma.animal.findMany()
-    });
+  fastify.get("/animals", async (req, res) => {
+    return fastify.prisma.animal.findMany();
+  });
 
-    fastify.get('/animals/:animalId', async (req, res) => {
-        const { animalId } = req.params;
-        return fastify.prisma.animal.findUnique({ where: { id: Number(animalId) } })
+  fastify.get("/animals/:animalId", async (req, res) => {
+    const { animalId } = req.params;
+    return fastify.prisma.animal.findUnique({
+      where: { id: Number(animalId) },
     });
+  });
+
+  const postBodyJsonSchema = {
+    type: "object",
+    required: ["animal"],
+    properties: {
+      animal: { type: "string" },
+    },
+  };
+
+  fastify.post("/animals", { body: postBodyJsonSchema }, async (req, res) => {
+    const { animal } = req.body;
+    return fastify.prisma.animal.create({ data: { name: animal } });
+  });
 
-    const postBodyJsonSchema = {
-        type: 'object',
-        required: ['animal'],
-        properties: {
-            animal: { type: 'string' },
-        }
-    };
+  fastify.put(
+    "/animals/:animalId",
+    { body: postBodyJsonSchema },
```

Comment:
nitpick: specifying the type as number will let the serializer[[ajv](https://github.com/ajv-validator/ajv)] handle the type conversion and any unexpected input ;)

```suggestion
    { 
        body: postBodyJsonSchema,
        params: {
            animaId: { type: "number" }
        }
     },
```